docs/Python-API.md

```diff
@@ -266,3 +268,172 @@ Once a property has been modified in Python, you can access it in C# after the n
+
+ * `channel_type(self) -> int` (property) : Must return an integer identifying the side channel (This number must
+be the same on C# and Python). There can only be one side channel of a certain type during communication.
+ * `on_message_received(self, data: bytes) -> None` : You must implement this method to specify what the
+ side channel will be doing with the data received from Unity. The data is a `byte[]` argument.
+
+To send a byte array from Python to C#, call the `super().queue_message_to_send(bytes_data)` method inside the
+side channel. The `bytes_data` argument must be a `bytes` object.
+
+To register a side channel on the Python side, call pass the side channel as argument when creating the
+`UnityEnvironment` object. One of the arguments of the constructor (`side_channels`) is a list of side channels.
+
+##### Example implementation
+
+Here is a simple implementation of a Side Channel that will exchange strings between C# and Python
+(encoded as ascii).
+
+One the C# side :
+Here is an implementation of a `StringLogSideChannel` that will listed to the `UnityEngine.Debug.LogError` calls in
+the game :
+
+```csharp
+using UnityEngine;
+using MLAgents;
+using System.Text;
+
+public class StringLogSideChannel : SideChannel
+{
+    public override int ChannelType()
+    {
+        return (int)SideChannelType.UserSideChannelStart + 1;
+    }
+
+    public override void OnMessageReceived(byte[] data)
+    {
+        var receivedString = Encoding.ASCII.GetString(data);
+        Debug.Log("From Python : " + receivedString);
+    }
+
+    public void SendDebugStatementToPython(string logString, string stackTrace, LogType type)
+    {
+        if (type == LogType.Error)
+        {
+            var stringToSend = type.ToString() + ": " + logString + "\n" + stackTrace;
+            var encodedString = Encoding.ASCII.GetBytes(stringToSend);
+            base.QueueMessageToSend(encodedString);
+        }
+    }
+}
+```
+
+We also need to register this side channel to the Academy and to the `Application.logMessageReceived` events,
+so we write a simple MonoBehavior for this. (Do not forget to attach it to a GameObject in the scene).
+
+```csharp
+using UnityEngine;
+using MLAgents;
+
+
+public class RegisterStringLogSideChannel : MonoBehaviour
+{
+
+    StringLogSideChannel stringChannel;
+    public void Awake()
+    {
+        // We create the Side Channel
+        stringChannel = new StringLogSideChannel();
+
+        // When a Debug.Log message is created, we send it to the stringChannel
+        Application.logMessageReceived += stringChannel.SendDebugStatementToPython;
+
+        // Just in case the Academy has not yet initialized
+        Academy.Instance.RegisterSideChannel(stringChannel);
+    }
+
+    public void OnDestroy()
+    {
+        // De-register the Debug.Log callback
+        Application.logMessageReceived -= stringChannel.SendDebugStatementToPython;
```

Comment:
Doesn't it need to deregister from the Academy too?

Reply:
I did not think we would need to deregister from the Academy. When/if the communicator is Disposed of, it will loose references to the SideChannels.

Reply:
In this case, maybe not. If the communication were the other way around, the Academy would still be referencing the (now destroyed) SideChannel, and would try to send to it if it received a message from python.

That's not relevant for this specific example, but if this is the template that people are going to copy, it should follow best practices.

Reply:
Academy does not reference the side channel, it directly passes it to the communicator

Reply:
Ah, you want to destroy the side channel but keep communicating. I see, then maybe I can add a Deregister side channel.

Reply:
It's still a valid point, isn't it? We should have a mechanism to deregister a side channel.

Reply:
Sorry, crossed message there. Yeah.

Reply:
I added such call